File: basic/extract variable/python/src/example_2.py
"""
Contexto: Estás desarrollando un programa para calcular el promedio de calificaciones de un estudiante en un curso. Tienes una función que recibe una lista de calificaciones y calcula el promedio. Actualmente, la suma de las calificaciones se encuentra en línea en el código. Aplica "Extract Variable" para mejorar la legibilidad del código al asignar la suma de las calificaciones a una variable con un nombre descriptivo.
"""

def calcular_promedio_(calificaciones):
    suma_calificaciones = sum(calificaciones)
    promedio = suma_calificaciones / len(calificaciones)
    return promedio

def calcular_promedio(calificaciones: list[float]):
    suma_calificaciones = sum(calificaciones)
    total_materias = len(calificaciones)
    promedio = suma_calificaciones / total_materias
    return promedio

File: basic/extract variable/javascript/example2.js
/*
Contexto: Estás desarrollando una aplicación de geometría. Tienes una función que calcula el área de un círculo basado en su radio. La fórmula del área del círculo se encuentra directamente en el código. Aplica "Extract Variable" para mejorar la legibilidad del código al asignar la fórmula del área a una variable con un nombre descriptivo.
*/
function redondear(numero, totalDecimales) {
    return numero.toFixed(totalDecimales)
}

function calcularAreaCirculo(radio) {
    let area = 3.14159265359 * radio * radio
    return redondear(area, 4)
}

function calcularAreaCirculoNuevo(radio) {
    const PI = 3.14159265359
    let radio2 = radio * radio
    let area = PI * radio2
    return redondear(area, 4)
}

export {calcularAreaCirculo, calcularAreaCirculoNuevo};